inc/class-wpseo-replace-vars.php

```diff
@@ -328,7 +328,8 @@ private function retrieve_date() {
 	private function retrieve_excerpt() {
 		$replacement = null;
 
-		if ( ! empty( $this->args->ID ) ) {
+		// The check `post_password_required` is because excerpt must be hidden for a post with a password.
+		if ( ! empty( $this->args->ID ) && ! post_password_required( $this->args->ID ) ) {
```

Comment:
Could you please add the `if` check from the following line, into this one as well (like you did for `retrieve_excerpt_only`) or turn this check into a simple function that can be reused?